h2/src/main/org/h2/fulltext/FullTextSettings.java

```diff
@@ -33,17 +35,17 @@
     /**
      * The set of words not to index (stop words).
      */
-    private final HashSet<String> ignoreList = New.hashSet();
+    private final Set<String> ignoreList = New.hashSet();
 
     /**
      * The set of words / terms.
      */
-    private final HashMap<String, Integer> words = New.hashMap();
+    private final Map<String, Integer> words = New.hashMap();
 
     /**
      * The set of indexes in this database.
      */
-    private final HashMap<Integer, IndexInfo> indexes = New.hashMap();
+    private final Map<Integer, IndexInfo> indexes = Collections.synchronizedMap(New.<Integer, IndexInfo>hashMap());
```

Comment:
This one is still needed, I think. Otherwise you end up with for sinle-line synchronized blocks elsewhere or have a race.